models/DataObject/AbstractObject.php

```diff
@@ -1399,28 +1399,32 @@ public static function __callStatic($method, $arguments)
     }
 
     /**
-     * @param  array  $listConfig
-     * @param  mixed $objectTypes
-     *
+     * @param array $listConfig
+     * @param mixed $objectTypes
      * @return Listing
-     *
      * @throws \Exception
      */
     protected static function makeList(array $listConfig, mixed $objectTypes): Listing
     {
         $list = static::getList($listConfig);
+        if(empty($objectTypes)) {
+            $objectTypes = [static::OBJECT_TYPE_VARIANT, static::OBJECT_TYPE_OBJECT];
+        }
 
-        // Check if variants, in addition to objects, to be fetched
-        if (!empty($objectTypes)) {
-            if (\array_diff($objectTypes, [static::OBJECT_TYPE_VARIANT, static::OBJECT_TYPE_OBJECT])) {
-                Logger::error('Class: DataObject\\AbstractObject => Unsupported object type in array ' . implode(',', $objectTypes));
+        if(!is_array($objectTypes)) {
+            Logger::error('Class: DataObject\\AbstractObject => Unsupported argument type ' . gettype($objectTypes)  . ' in function ' . __FUNCTION__);
 
-                throw new \Exception('Unsupported object type in array [' . implode(',', $objectTypes) . '] in class DataObject\\AbstractObject');
-            }
+            throw new \Exception('Argument $objectTypes must be of type array, ' .  gettype($objectTypes) . ' given.');
+        }
```

Comment:
What's wrong with changing the parameter type? It's way easier than doing the error handling manually.